### PR TITLE
GPC-NONE: Move platform into subdirectory to add test lambda

### PR DESCRIPTION
**Pull request by oskarwilliams**

### CODE REVIEW

File: lambdas/test-lambda/src/test/java/lep/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package lep;
```
Comment (olevett) [inline lambdas/test-lambda/src/test/java/lep/MainTest.java:1]: Is there a reason this is in `lep`, not `uk.gov.di.data.lep`?

File: platform/src/test/kotlin/uk/gov/gdx/datashare/e2e/AcquirerTest.kt [hunk 2]
```diff
@@ -1,9 +1,9 @@
-package uk.gov.gdx.datashare.uk.gov.gdx.datashare.e2e
+package uk.gov.gdx.datashare.e2e
```
Comment (olevett) [inline platform/src/test/kotlin/uk/gov/gdx/datashare/e2e/AcquirerTest.kt:1]: Ideally, I'd have done this as one change, then the other bits as a later change, this is sort of unrelated to the build layout changes

File: platform/src/test/resources/uk/gov/gdx/datashare/integration/api/approvals/EventsQueryTest.getEvent returns a death notification event in the correct format.approved.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (olevett) [inline platform/src/test/resources/uk/gov/gdx/datashare/integration/api/approvals/EventsQueryTest.getEvent returns a death notification event in the correct format.approved.json:1]: Why is this a new file vs a moved one? Does it duplicate something?

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    id "java"
+}
+
+repositories {
+    mavenCentral()
+}
+
+ext {
+    dependencyVersions = [
+            awsLambdaJavaCore:'1.2.1'
```
Comment (olevett) [inline build.gradle:11]: Are these going to be covered well by dependabot?

File: lambdas/test-lambda/build.gradle [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    id "java"
+    id "jacoco"
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation "com.amazonaws:aws-lambda-java-core:$rootProject.ext.dependencyVersions.awsLambdaJavaCore"
```
Comment (olevett) [inline lambdas/test-lambda/build.gradle:11]: If we're using a shared versioning, I'd be keen to apply it to all deps, not just this one

**Comment by oskarwilliams** [comment]
Made this draft as I am splitting it up into smaller tickets

**Pull request closed without merging**